spec/Appendix B -- Grammar Summary.md

```diff
@@ -172,3 +173,57 @@ NonNullType :
 Directives : Directive+
 
 Directive : @ Name Arguments?
```

Comment:
I think this needs to express that argument values are constants for directives used in definitions.  The following is allowed by the grammar right now, I believe.
```
scalar foo @deprecated(reason: $foo)
```
Unless I'm missing something, there's no enclosing `VariableDefinitions` for schema to define variables, so this doesn't seem to make much sense.

The arguments definitions are:
```
Arguments : ( Argument+ )
Argument : Name : Value
Value[Const] :
```

I think in this case, it should be something like:

```
Directive : @ Name Arguments[Const]?
```
That involves also changing `Arguments` and `Argument`, right?  I'm not sure what the right thing is here, but I think the current definition is ambiguous.

Reply:
(Sorry for delayed feedback)

You're absolutely correct. Variables should not be allowed by the grammar. I will account for that